src/dotnet/APIView/APIViewWeb/Controllers/PullRequestController.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using APIViewWeb.Repositories;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System.Threading.Tasks;
+
+namespace APIViewWeb.Controllers
+{
+    public class PullRequestController : Controller
+    {
+        private readonly PullRequestManager _pullRequestManager;
+        private readonly ILogger _logger;
+
+        public PullRequestController(PullRequestManager pullRequestManager, ILogger<AutoReviewController> logger)
+        {
+            _pullRequestManager = pullRequestManager;
+            _logger = logger;
+        }
+                
+        [HttpGet]
+        public async Task<ActionResult> DetectApiChanges(
+            string buildId, 
+            string artifactName, 
+            string filePath, 
+            int pullRequestNumber, 
+            string commitSha,
+            string repoName)
```

Comment:
Do you not still need the language in order to know what processing or artifact types to look for?

Reply:
Nope. Each language is passing the path to artifact which is required as filter path in download request. And this path is discovered by our existing code in Language-settings script to identify artifact for API review system. API review doesn't need language since we parse that info out of Artifact when stub file is generated.